src/firefly/js/visualize/ui/CatalogTableListField.jsx

```diff
@@ -0,0 +1,227 @@
+     * @param np
+     * @returns {boolean}
+     */
+    shouldComponentUpdate(np) {
+        const newCatValue = get(np, 'value', '');
+        const oldCatValue = get(this.props, 'value', '');
+        //FIXME Doesn't work when user comes back from another tab
+        return newCatValue != oldCatValue;
+    }
+
+    render() {
+        var {data, cols, onClick, fieldKey} = this.props;//data {cat:...}
+        const indexClicked = get(this.state, 'idx', 0);
+
+        if (isEmpty(data)) {
+            return (
+                <div style={{position: 'relative'}}>
+                    <div className='loading-mask'/>
+                </div>
+            );
+        }
+
+        var items = data.map((o, index) => {
+
+
+            let clicked = false;
+            if (indexClicked === index) {
+                clicked = true;
+            }
+            return (
+
+                <CatalogTableItem
+                    key={index}
+                    onClick={onClick}
+                    isClicked={clicked}
+                    cols={cols}
+                    itemData={o}/>
+            );
+        });
+
+        return (
+            <div>
+                <table name={fieldKey}>
+                    <tbody>
+                    {items}
+                    </tbody>
+                </table>
+
+            </div>
+        );
+    }
+}
+
+export const CatalogTableListField = fieldGroupConnector(CatalogTableView, getProps, CatalogTableView.propTypes, null);
+
+CatalogTableView.propTypes = {
+    data: PropTypes.array.isRequired,
+    cols: PropTypes.array.isRequired,
+    groupKey: PropTypes.string,
+    onClick: PropTypes.func,
+    fieldKey: PropTypes.string
+};
+
+CatalogTableView.defaultProps = {
+    data: [],
+    cols: []
+};
+
+
+function getProps(params, fireValueChange) {
+
+    return Object.assign({}, params,
+        {
+            onClick: (ev) => handleOnClick(ev, params, fireValueChange)
+        });
+}
+
+function handleOnClick(ev, params, fireValueChange) {
+
+    // the value of this input field is a string
+    fireValueChange({
+        value: ev.currentTarget.value || ev.currentTarget.attributes['value'].value
+    });
+}
+
+// get element data and wrapit around html markup knowing the columns name
+function htmlElementMarkup(element, cols) {
+    let m = '';
+    const url = '<a target=\'_new\' href=\'http://irsa.ipac.caltech.edu/cgi-bin/Gator/nph-dd?mode=xml&catalog=@\'>go to dd</a>';
+    cols.forEach((e, idx) => {
+            if (idx < 10) {
+                //<span class="marked-text" style="font-size:80%;">Rows: </span><span class="normal-text" style="font-size:80%;">18,240</span>
+                m += '<b>' + e.name + '</b>: ' + element.cat[idx];
+                if (e.name.includes('cat')) {
+                    m += ' [ ' + url.replace('@', element.cat[idx]) + ' ]';
+                }
+                m += '    ';
+            }
+        }
+    );
+    return m;
+}
+
+class CatalogTableItem extends Component {
```

Comment:
same comment here.  if you do not need life cycle methods, use functional component.